cookbooks/aws-parallelcluster-slurm/recipes/init_dns.rb

```diff
@@ -96,7 +96,21 @@
 # Configure fqdn in /etc/hosts
 replace_or_add "set fqdn in the /etc/hosts" do
   path "/etc/hosts"
-  pattern "^#{node['ec2']['local_ipv4']}\s+"
-  line(lazy { "#{node['ec2']['local_ipv4']} #{node['cluster']['assigned_hostname'].chomp('.')} #{node['cluster']['assigned_short_hostname']}" })
+  primary_ip = ""
+  token = get_metadata_token
+  macs = network_interface_macs(token)
+  for mac in macs
+    uri = URI("http://169.254.169.254/latest/meta-data/network/interfaces/macs/#{mac}/device-number")
+    device_number = get_metadata_with_token(token, uri)
+    uri = URI("http://169.254.169.254/latest/meta-data/network/interfaces/macs/#{mac}/network-card")
+    network_card = get_metadata_with_token(token, uri)
+    next unless device_number == '0' && network_card == '0'
+
+    uri = URI("http://169.254.169.254/latest/meta-data/network/interfaces/macs/#{mac}/local-ipv4s")
+    primary_ip = get_metadata_with_token(token, uri)
+    break
+  end
+  pattern "^#{primary_ip}\s+"
```

Comment:
**Thought:** what happens if the ip already present in the `/etc/hosts` is not what we identify as `primary_ip`. I think that we should clear each line where we find an IP associated with the current instance

Reply:
I'm not sure that I completely understand your question. If the IP is already correct then there wouldn't be any issue. So, what do you mean by "if the ip is already present?"